ESP3Parser.js

```diff
@@ -2,160 +2,112 @@
 "use strict";
 
 // Copyright 2015 Rafael Kähm <rafael@kaehms.de>
+// Copyright 2018 Holger Will <h.will@klimapartner.de>
 
+const Transform = require('stream').Transform
+const ESP3Packet = require('./ESP3Packet');
 
-var ESP3Packet = require('./ESP3Packet');
-var esp3 = new ESP3();
 
 // Emit a data event by recognizing ESP3 packets
 // Data contains ESP3Packet
 // More: https://www.enocean.com/fr/knowledge-base-doku/enoceansystemspecification:issue:faqesp2esp3/?purge=1
-module.exports = function (emitter, buffer) {
-	esp3.tryToFillPackets(emitter, buffer);
-};
-
-function ESP3() {
-	var currentESP3Packet;
-	var emitter;
-
-	var tmp = {
-	};
-
-	var callbackForNextByte = waitForSyncByte;
-
-	this.tryToFillPackets = function(emitterU, buffer) {
-		emitter = emitterU;
-		for (var offset = 0; offset < buffer.length; offset++) {
-			callbackForNextByte(buffer[offset]);
+class EnoceanParser extends Transform {
+	constructor(options = {}) {
+		super(options)
```

Comment:
Suggestion 1 : Keep "Object Mode"  
```suggestion
		super(Object.assign(options, {readableObjectMode: true}));
```